### ISSUE TITLE
trailing slash in route stripped regardless of LooseSlash option

### ISSUE THREAD

**Issue by coreyk** [gobuffalo/buffalo#955]
App with these routes:
```
app.GET("/top-comments-abc", TopCommentList)
app.GET("/top-comments-abc/", TopCommentList)
```

and this logging in router.go:

```
func (a *App) addRoute(method string, url string, h Handler) *RouteInfo {
    a.moot.Lock()
    defer a.moot.Unlock()
    l := log.New(os.Stdout, "", 0)
    l.Println("router url : " + url)

    url = path.Join(a.Prefix, url)
    name := a.buildRouteName(url)

    l.Println("router url post join : " + url)
    l.Println("router name : " + name)
```

prints this on build:

```
router url : /
router url 2 : /
router name : root
router url : /top-comments-abc
router url post join : /top-comments-abc
router name : topCommentsAbc
router url : /top-comments-abc/
router url post join : /top-comments-abc
router name : topCommentsAbc
```

which suggests that 

``` 
url = path.Join(a.Prefix, url)
```

 is removing the trailing slash.

Sometimes that trailing slash is necessary.

**Comment by markbates** [comment]
I'm struggling to build test cases properly, would you mind providing a map of expected outcomes? something like:

| mapped url | path given | expected status |
| --- | --- | --- |
| `/top-comments/` | `/top-comments/` | 200 |
| `/top-comments/` | `/top-comments` | 400 |

**Comment by coreyk** [comment]
When using LooseSlash: false, that is correct.

| mapped url | path given | expected status |
| --- |---|---|
| /post/{id} | /post/10/ | 404 |
| /user/{name}/ | /user/joe |   404 |
| /comment/{id}/ | /comment/342/ | 200 |
| /users/{id}/edit | /users/2/edit/ | 404 |

**Comment by stanislas-m** [comment]
Fixed with #1168.
